File: Examples/Java/SDK/src/main/java/com/aspose/cells/cloud/examples/cells/GetMinDataRowWorksheet.java
package com.aspose.cells.cloud.examples.cells;

import com.aspose.cells.api.CellsApi;
import com.aspose.cells.cloud.examples.Configuration;
import com.aspose.cells.cloud.examples.Utils;
import com.aspose.storage.api.StorageApi;

import java.io.IOException;
import java.nio.file.Path;

public class GetMinDataRowWorksheet {

    public static void main(String... args) throws IOException {
        // ExStart: get-min-data-row
        try {
            // Instantiate Aspose Storage API SDK
            StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);

            // Instantiate Aspose Words API SDK
            CellsApi cellsApi = new CellsApi(Configuration.apiKey, Configuration.appSID, true);
            String input = "sample1.xlsx";
            Path inputFile = Utils.getPath(GetMinDataRowWorksheet.class, input);

            String sheetName = "Sheet1";
            String cellOrMethodName = "mindatarow";

            storageApi.PutCreate(input, null, Utils.STORAGE, inputFile.toFile());

            Integer apiResponse = cellsApi.GetWorksheetCellProperty(input, sheetName, cellOrMethodName, Utils.STORAGE,
                    Utils.FOLDER);
            System.out.println(" MaxDataRow :: " + apiResponse);
        }

        catch (Exception e) {
            e.printStackTrace();
        }
        // ExEnd: get-min-data-row

    }
}

File: Examples/Perl/Cells/GetMinDataRowWorksheet.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;
use AsposeCellsCloud::CellsApi;
use AsposeCellsCloud::ApiClient;
use AsposeCellsCloud::Configuration;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeCellsCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeCellsCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeCellsCloud::Configuration::debug = 1;
$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Cells API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $cellsApi = AsposeCellsCloud::CellsApi->new();

# Set input file name
my $name = 'Sample_Test_Book.xls';
my $sheetName = 'Sheet2';
my $cellOrMethodName = "mindatarow";

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);
# Invoke Aspose.Cells Cloud SDK API to get index of first row which contains data in a worksheet                     
$response = $cellsApi->GetWorksheetCell2(name=> $name, sheetName=>$sheetName, cellOrMethodName=>$cellOrMethodName);
print "\nMinDataRow :: " . $response->content;
#ExEnd:1